### ISSUE TITLE
how can fix "ips" when im in loophole 

### ISSUE THREAD

**Issue by derzbot2002** [UnamSanctam/UnamWebPanel#80]
![Captura](https://user-images.githubusercontent.com/98950427/152325736-01fbcd0d-a9ce-4c53-91f6-945af759feb1.PNG)

Hello, I need help, how can i fix the ips, im using loophole "portmap.io"
it only recreates the ip all the time

**Comment by UnamSanctam** [comment]
Well that's the IP that the traffic comes from, is it even possible to know what the origin IP is when using portmap (or any loophole)?

**Comment by derzbot2002** [comment]
Im using openvpn connected with portmap file with my config

**Comment by UnamSanctam** [comment]
Yes and 10.9.0.1 is probably the local IP that the connections are coming from which is why it shows that. It gets the address using $_SERVER["REMOTE_ADDR"] which in this case is coming from your local connection, it's the same if using Cloudflare since it proxies all the traffic so for Cloudflare you have to check the http header instead.

**Comment by derzbot2002** [comment]
so what i should do ? :)

**Comment by UnamSanctam** [comment]
You would have to set up that with portmap if it's even possible, you can't do anything in the PHP to "fix" it (from as far as I know) since the PHP receives everything from 10.9.0.1 and probably does not even know the real IPs. I haven't used portmap before so I'm not sure if there's some setting or configuration in OpenVPN that can expose the real IPs.

**Comment by derzbot2002** [comment]
ok thank you

**Comment by grandepremo** [comment]
I have a similar problem, only for each miner, it shows the IP of the server on which the panel is located instead of the IP of the miner

**Comment by UnamSanctam** [comment]
I don't know your setup but if you're hosting it behind a proxy that routes all the traffic through it then it will show that proxy IP since that's where the traffic is coming from, your webserver can't know what the original IP was unless it's somehow told by the proxy.
